docs/flops.md

```diff
@@ -0,0 +1,188 @@
+## FLOP counting with NATTEN
+
+NATTEN offers multiple solutions for counting FLOPs/MACs.
+
+However, note that due to the variety of solutions available for this, it is important to have a
+consistent definitions of FLOPs and how they are counted.
+
+### Rules
+Since NATTEN is limited to the dot product attention operator, here we note the general rules for
+counting FLOPs/MACs in NATTEN:
+
+Following standard practice, only Tensor Core (TC) FLOPs are counted.
+Softmax, masking, attention scale, and positional biases are NOT included in the FLOP/MAC count.
+
+For a GEMM problem with shape `(M, N, K)`, MACs (Multiply-and-Accumulate operations) is simply
+`M * N * K`, while FLOPs (Floating Point Operations) considers multiply and accumulate separate
+operations, and ends up being `2 * M * N * K`.
+
+Causal masking also does **not** affect FLOPs/MACs today, but this behavior may change in the
+future. It makes sense to measure in causal masking for various reasons, but the actual FLOP
+decrease is not so granular in practice. More details will be added about this in the future.
+Please refer to [#184](https://github.com/SHI-Labs/NATTEN/issues/184#issuecomment-2505022903) for
+more information.
+
+### Manual
+All users can integrate NATTEN's FLOP/MAC counting logic into their own FLOP counting logic:
+
+```python
+from natten.flops import (
+  fna_flop_count,
+  na_qk_flop_count,
+  na_av_flop_count
+)
+```
+
+If you're using [Fused Neighborhood Attention (FNA)](fna/), or just prefer to compute FLOPs/MACs for
+the entire attention operator all at once, use `fna_flop_count`:
+
+```python
+fna_flop_count(
+    q_shape: torch.Size | Sequence[int] | List[int],
```

Comment:
it probably suffices to simplify this to `Sequence[int]` only. I believe List is a type of Sequence already, and torch.Size is a Tuple, thus a Sequence too. an (input) interface is as wide as its widest type, so when unioning input parameter types, we should just state the widest if any share superclasses.

can do this in flops.py too.